pkg/providers/amifamily/ami.go

```diff
@@ -181,13 +214,13 @@ func (p *Provider) fetchAMIsFromSSM(ctx context.Context, ssmQuery string) (strin
 	return ami, nil
 }
 
-func (p *Provider) getAMIsFromSelector(ctx context.Context, nodeTemplate *v1alpha1.AWSNodeTemplate) ([]AMI, error) {
-	ec2AMIs, err := p.fetchAMIsFromEC2(ctx, nodeTemplate.Spec.AMISelector)
+func (p *Provider) getAMIsFromSelector(ctx context.Context, selector map[string]string) ([]AMI, error) {
+	ec2AMIs, err := p.fetchAMIsFromEC2(ctx, selector)
 	if err != nil {
 		return nil, err
 	}
 	if len(ec2AMIs) == 0 {
-		return nil, fmt.Errorf("no amis exist given constraints")
+		logging.FromContext(ctx).Errorf("no amis exist given constraints")
```

Comment:
We still aren't returning an error here. From discussion offline, I was under the impression that we were going to be returning errors now if we don't resolve anything.